src/status_im/multiaccounts/recover/core.cljs

```diff
@@ -127,9 +137,159 @@
 (fx/defn navigate-to-recover-multiaccount-screen [{:keys [db] :as cofx}]
   (fx/merge cofx
             {:db (dissoc db :multiaccounts/recover)}
-            (navigation/navigate-to-cofx :recover nil)))
+            (navigation/navigate-to-cofx :recover-multiaccount nil)))
 
 (re-frame/reg-fx
  :multiaccounts.recover/recover-multiaccount
  (fn [[masked-passphrase password]]
    (recover-multiaccount! masked-passphrase password)))
+
+(re-frame/reg-fx
+ :multiaccounts.recover/import-mnemonic
+ (fn [{:keys [passphrase password]}]
+   (status-im.native-module.core/multiaccount-import-mnemonic
+    passphrase
+    password
+    (fn [result]
+      (re-frame/dispatch [:multiaccounts.recover/import-mnemonic-success result])))))
+
+(re-frame/reg-fx
+ :multiaccounts.recover/derive-addresses
+ (fn [{:keys [account-id paths]}]
+   (status-im.native-module.core/multiaccount-derive-addresses
+    account-id
+    paths
+    (fn [result]
+      (re-frame/dispatch [:multiaccounts.recover/derive-addresses-success result])))))
+
+(fx/defn on-import-mnemonic-success
+  {:events [:multiaccounts.recover/import-mnemonic-success]}
+  [{:keys [db] :as cofx} result]
+  (let [{:keys [id]} (types/json->clj result)]
+    {:multiaccounts.recover/derive-addresses {:account-id id
+                                              :paths      [constants/path-default-wallet]}}))
```

Comment:
@dmitryn you have to derive whisper as well, for multiaccounts we have different paths for whisper and for wallet

Reply:
@flexsurfer yes, but what key we show on "Successfully recovered" screen? https://github.com/status-im/status-react/pull/8687#issuecomment-517351780
Now we show key derived from wallet path.

Reply:
we should show whisper public key, you need to add to paths constant/whisper and get public key from whisper path keyword, not wallet

Reply:
it should work same as here https://github.com/status-im/status-react/blob/60de46b6c22aa8266a6739b7c730cb7337af7e05/src/status_im/multiaccounts/create/core.cljs#L290

Reply:
I mean this derivation is only used to show user what key he recovered

Reply:
so this is how it should work
1) import mnemonic -> get id and root address
2) derive whisper paths and wallet path using id
3) show whisper public key
4) create new multiaccount with root address, whisper public key and default wallet account (address)

Reply:
also you need to change `validate-recover-result`

Reply:
@flexsurfer i'm using status-go `recoverAccount(passphrase, password)`, should i use `multiAccountStoreDerived` instead?

Reply:
This one https://github.com/status-im/status-react/blob/60de46b6c22aa8266a6739b7c730cb7337af7e05/src/status_im/multiaccounts/create/core.cljs#L49

Reply:
yes you should use `multiAccountStoreDerived`

Reply:
@flexsurfer requests changes applied, please take a look